Grundgeruest/views.py

```diff
@@ -274,18 +273,20 @@ def paypal_bestaetigung(request):
 
 def upgrade_nutzer(request, datendict):
     ''' Setzt die neue Unterstützerstufe nach erfolgreicher Zahlung.'''
-    # FIXME: Höhere Stufe kann durch kleinere verlängert werden! Unterstützungsmodel implementieren stattdesssen!
     if request.user.is_authenticated():
-        nutzer = request.user
+        profile = request.user.my_profile
     else:
-        nutzer = Nutzer.objects.get(email=datendict['email'])
-    if nutzer.my_profile.stufe < int(datendict['stufe']):
-        nutzer.my_profile.stufe = int(datendict['stufe'])
-    nutzer.my_profile.guthaben_aufladen(int(datendict['betrag']))
-    today = datetime.now().date()
-    nutzer.my_profile.letzte_zahlung = today
-    nutzer.my_profile.datum_ablauf = today + timedelta(days=365)
-    nutzer.my_profile.save()
+        profile = Nutzer.objects.get(email=datendict['email']).my_profile
+
+    unterstuetzung = Unterstuetzung(profil=profile,
+                                    stufe=Spendenstufe.objects.get(pk=datendict['stufe']),
+                                    datum=date.today(),
+                                    zahlungsmethode=datendict['zahlungsweise'])
+    unterstuetzung.save()
+
+    profile.guthaben_aufladen(int(datendict['betrag']))

```

Comment:
hm, gehört das nicht eher ins Unterstuetung.__init__?
bin nicht sicher, was dagegen spricht; wäre doch more pythonic.

Reply:
__init__ sollte nicht überschriben werden nach django. ich schreibst in save() falls noch keine pk da ist.